pdbtools/pdb_tidy.py

```diff
@@ -62,25 +64,52 @@ def check_input(args):
             sys.exit(1)
 
     elif len(args) == 1:
-        if not os.path.isfile(args[0]):
-            emsg = 'ERROR!! File not found or not readable: \'{}\'\n'
+        # One of two options: option & Pipe OR file & default option
+        if args[0] == '-strict':
+            option = True
+            if sys.stdin.isatty():  # ensure the PDB data is streamed in
+                emsg = 'ERROR!! No data to process!\n'
+                sys.stderr.write(emsg)
+                sys.stderr.write(__doc__)
+                sys.exit(1)
+
+        else:
+            if not os.path.isfile(args[0]):
+                emsg = 'ERROR!! File not found or not readable: \'{}\'\n'
+                sys.stderr.write(emsg.format(args[0]))
+                sys.stderr.write(__doc__)
+                sys.exit(1)
+
+            fh = open(args[0], 'r')
+
+    elif len(args) == 2:
+        # Two options: option & File
+        if not args[0] == '-strict':
+            emsg = 'ERROR! First argument is not a valid option: \'{}\'\n'
             sys.stderr.write(emsg.format(args[0]))
             sys.stderr.write(__doc__)
             sys.exit(1)
 
-        fh = open(args[0], 'r')
+        if not os.path.isfile(args[1]):
+            emsg = 'ERROR!! File not found or not readable: \'{}\'\n'
+            sys.stderr.write(emsg.format(args[1]))
+            sys.stderr.write(__doc__)
+            sys.exit(1)
+
+        option = args[0][1:]
+        fh = open(args[1], 'r')
 
     else:  # Whatever ...
-        emsg = 'ERROR!! Script takes 1 argument, not \'{}\'\n'
-        sys.stderr.write(emsg.format(len(args)))
         sys.stderr.write(__doc__)
         sys.exit(1)
 
-    return fh
+    return (option, fh)
 
 
-def tidy_pdbfile(fhandle):
+def tidy_pdbfile(fhandle, strict=False):
     """Adds TER/END statements and pads all lines to 80 characters.
+
+    If strict is True, does not add TER statements at intra-chain breaks.
     """
 
```

Comment:
Create a variable with the inverse of the `bool` value so to avoid the need to compute `not strict` every iteration of the loop.

```python
not_strict = not strict
```

Reply:
Done.